lib/galaxy/tools/parameters/output_collect.py

```diff
@@ -164,6 +164,9 @@ def collect_dynamic_collections(
         else:
             collection = has_collection
 
+        # We are adding dynamic collections, which may be precreated, but their actually state is still new!
+        collection.populated_state = collection.populated_states.NEW
```

Comment:
This seems wrong, this piece of the collection is about to be materialized. Maybe the rest of the collection has new states but this one should have a populated state - at least by the end of this method.